iothub/device/src/Fx/Fx.cs

```diff
@@ -27,7 +27,7 @@ internal static bool IsFatal(Exception ex)
 
             if (ex is IotHubClientException hex)
             {
-                return hex.IsTransient;
+                return !hex.IsTransient;
```

Comment:
`IotHubClientException` whose `IsTransient` flag is "true" should **not** be marked as Fatal.